File: algorithms/Fibonacci Number/Fibonacci Number.java
class Solution {
    public int fib(int n) {
        
        int[] fiboArray = new int[n+2];
        
        fiboArray[0] = 0;
        fiboArray[1] = 1;
        
        for(int i=2; i<=n; i++) {
            fiboArray[i] = fiboArray[i-1] + fiboArray[i-2];
        }
        
        return fiboArray[n];
        
    }
}

File: scripts/algorithms/F/Fibonacci Number/Fibonacci Number.cpp
class Solution {
public:
    int fib(int n) {
        double sqrt5 = 2.23606797749979;
        double expr1 = 1.61803398874989;
        double expr2 = -0.618033988749895;
        double res = (pow(expr1,n)-pow(expr2,n))/sqrt5;
        int result = (int)round(res);
        return result;
    }
};
